mapie/calibration.py

```diff
@@ -0,0 +1,548 @@
+                item,
+                calibrator,
+                y,
+                max_prob,
+                y_pred,
+                sample_weight,
+            )
+            calibrators[item] = calibrator_
+        return calibrators
+
+    def _pred_proba_calib(
+        self,
+        idx: int,
+        item: int,
+        calibrated_values: NDArray,
+        max_prob: NDArray,
+        y_pred: NDArray,
+        calibrators: Dict[int, RegressorMixin],
+    ) -> NDArray:
+        """
+        Using the predicted probabilities, we calibrate new values with
+        the fitted calibrators. Note that if there is no calibrator for a
+        the specific class, then we simply output the not calibrated values.
+
+        Note that if the calibrated probability prediction is 0, there would
+        be an issue when finding the class is belongs to. We set it equal to
+        0.00001 as this would likely not have large impact on the calibration
+        scores, yet, would set the maximum to the correct label.
+
+
+        Parameters
+        ----------
+        item : int
+            The class value to be calibrated.
+        calibrated_values : NDArray
+            Array of calibrated values.
+        max_prob : NDArray of shape (n_samples,)
+            Values to be calibrated.
+        y_pred : NDArray of shape (n_samples,)
+            Indicator of the values to be calibrated.
+        calibrators : Dict[RegressorMixin] of len n_classes in calibration set
+            Dictionary of all the previously fitted calibrators.
+
+        Returns
+        -------
+        NDArray
+            Updated calibrated values from the predictions of the calibrators.
+
+        Warnings
+            If there has not been a fitted calibrator for the specific class.
+        """
+        correct_label = np.where(y_pred.ravel() == item)[0].ravel()
+        if item not in calibrators:
+            calibrated_values[
+                correct_label, idx
+                ] = max_prob[correct_label].ravel()
+            warnings.warn(
+                "WARNING: This predicted label has not been seen during the"
+                + " calibration and therefore scores will remain unchanged."
+            )
+        else:
+            EPSILON = 0.00001
+            calibrator_ = calibrators[item]
+            preds_ = calibrator_.predict(max_prob[correct_label])
+            idx_zero_pred = np.where(preds_ < EPSILON)[0]
+            preds_[idx_zero_pred] = EPSILON
+            calibrated_values[correct_label, idx] = preds_
+        return calibrated_values
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[NDArray] = None,
+        calib_size: Optional[float] = 0.33,
+        random_state: Optional[Union[int, np.random.RandomState, None]] = None,
+        shuffle: Optional[bool] = True,
+        stratify: Optional[ArrayLike] = None,
+    ):  # MapieCalibrator... Don't understand why it's not recognized?
```

Comment:
use 
from __future__ import annotations

Reply:
Thank you!